### PR TITLE
Improve convertToUnIndexedMesh and convertToFlatShadedMesh vertex data handling with support for morph targets

### PR DESCRIPTION
**Pull request by alexchuber**
Both convertToUnIndexedMesh() and convertToFlatShadedMesh() appear to be doing similar things, so I created a private function, _convertToUnIndexedMesh(), to be used by both.

_convertToUnIndexedMesh() has a flag to decide if normals should be flattened, as should be the case for flat shading. 
I also added logic to accomodate morph targets, if they're available in the mesh.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Meshes/mesh.ts
Comment (alexchuber) [inline packages/dev/core/src/Meshes/mesh.ts:3112]: I noticed that both of the original methods updated the index buffer. I'm not sure why, although this forum post gives some clue.
https://forum.babylonjs.com/t/mesh-converttounindexedmesh-fails-to-remove-indices/13269/2?u=alexchuber

Should the index buffer be updated every time? Or should the indices not be updated at all?
(I have it under a flag for now.)

**Comment by alexchuber** [comment]
Should this have a unit test? Visualization test?

File: packages/dev/core/src/Meshes/mesh.ts
Comment (bghgary) [inline packages/dev/core/src/Meshes/mesh.ts:3007]: Is this necessary? Lint shouldn't be looking at private functions.

File: packages/dev/core/src/Meshes/mesh.ts
Comment (bghgary) [inline packages/dev/core/src/Meshes/mesh.ts:3015]: I know I helped you write this part, but it could be further simplified:
```suggestion
        const flipNormalGeneration = this.overrideMaterialSideOrientation === (this._scene.useRightHandedSystem ? Constants.MATERIAL_CounterClockWiseSideOrientation : Constants.MATERIAL_ClockWiseSideOrientation);
```

File: packages/dev/core/src/Meshes/mesh.ts
Comment (bghgary) [inline packages/dev/core/src/Meshes/mesh.ts:3047]: ditto

File: packages/dev/core/src/Meshes/mesh.ts
Comment (bghgary) [inline packages/dev/core/src/Meshes/mesh.ts:3085]: nit:
```suggestion
        // Update morph targets
```

File: packages/dev/core/src/Meshes/mesh.ts
Comment (bghgary) [inline packages/dev/core/src/Meshes/mesh.ts:3053]: nit: How about this?
```suggestion
        const separateVertices = (data: FloatArray, stride: number): Float32Array => {
```

File: packages/dev/core/src/Meshes/mesh.ts
Comment (bghgary) [inline packages/dev/core/src/Meshes/mesh.ts:3112]: nit: style
```suggestion
        if (updateIndices) {
```

File: packages/dev/core/src/Meshes/mesh.ts
Comment (bghgary) [inline packages/dev/core/src/Meshes/mesh.ts:3144]: nit: extra newline
```suggestion
    }

    /**
```

File: packages/dev/core/src/Meshes/mesh.ts
Comment (bghgary) [reply packages/dev/core/src/Meshes/mesh.ts:3112]: @Popov72 @sebavan @deltakosh Thoughts?

File: packages/dev/core/src/Meshes/mesh.ts
Comment (alexchuber) [reply packages/dev/core/src/Meshes/mesh.ts:3015]: Sure!

File: packages/dev/core/src/Meshes/mesh.ts
Comment (bghgary) [reply packages/dev/core/src/Meshes/mesh.ts:3110]: Also, we shouldn't merge it as is. If we are going to have a flag, it should be exposed to the public function.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Review by Popov72 — APPROVED** [review]
Nice!

File: packages/dev/core/src/Meshes/mesh.ts
Comment (Popov72) [reply packages/dev/core/src/Meshes/mesh.ts]: I think we should keep backward compatibiliy, so keep the index array with 0,1,2,3,... values.

**Comment by Popov72** [comment]
> Should this have a unit test? Visualization test?

Maybe that would be nice to have a visualization test with the alien head and use `convertToFlatShadedMesh` on all meshes?

**Review by RaananW — APPROVED** [review]
LGTM!

Note - our UMD system compiles to ES5, so some es2015 features (and up) won't transpile correctly. This is the reason the build fails. If you want to quickly test is, run `npm run build:babylonjs` which will build core for UMD. you will see the error.

On top of that - please run `npm run format:fix` to fix all formatting issues.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14002/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14002/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14002/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14002/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14002/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14002/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14002/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14002/merge#BCU1XR#0

**Review by bghgary — APPROVED** [review]

**Review by sebavan — APPROVED** [review]
LGTM, as long as back and agree with @Popov72 on back compat.

**Pull request merged**
